tests/test_addDistortionModel.py

```diff
@@ -0,0 +1,124 @@
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+from __future__ import absolute_import, division, print_function
+import unittest
+
+import lsst.utils.tests
+import lsst.afw.geom as afwGeom
+from lsst.afw.coord import IcrsCoord
+from lsst.afw.image import ExposureF, ExposureInfo, Calib, VisitInfo
+from lsst.afw.geom import makeCdMatrix, makeSkyWcs
+from lsst.afw.geom.wcsUtils import makeDistortedTanWcs
+from lsst.afw.cameraGeom import FIELD_ANGLE, FOCAL_PLANE, PIXELS
+from lsst.afw.cameraGeom.testUtils import DetectorWrapper, CameraWrapper
+from lsst.afw.geom.utils import wcsAlmostEqualOverBBox
+from lsst.ip.isr import IsrTask
+
+
+class ApplyLookupTableTestCase(lsst.utils.tests.TestCase):
+    """Test IsrTask.addDistortionModel
+    """
+
+    def setUp(self):
+        self.camera = CameraWrapper().camera
+        self.detector = DetectorWrapper().detector
+        self.crpix = afwGeom.Point2D(50, 100)
+        self.crval = IcrsCoord(36 * afwGeom.degrees, 71 * afwGeom.degrees)
+        scale = 1.0 * afwGeom.arcseconds
+        self.cdMatrix = makeCdMatrix(scale=scale)
+        self.wcs = makeSkyWcs(crpix=self.crpix, crval=self.crval, cdMatrix=self.cdMatrix)
+        self.bbox = afwGeom.Box2I(afwGeom.Point2I(-10, 10), afwGeom.Extent2I(1000, 1022))
+        self.exposure = ExposureF(self.bbox)
+
+        # set the few items of ExposureInfo needed by IsrTask.run
+        # when only adding a distortion model
+        exposureInfo = ExposureInfo(calib = Calib(1.0),
+                                    detector = self.detector,
+                                    visitInfo = VisitInfo(exposureTime=1.0),
+                                    wcs = self.wcs)
+
+        self.exposure.setInfo(exposureInfo)
+
+    def tearDown(self):
+        self.detector = None
+        self.exposure = None
+
+    def testAddDistortionMethod(self):
+        """Call IsrTask.addDistortionModel directly"""
+        isrTask = IsrTask()
+        isrTask.addDistortionModel(self.exposure, self.camera)
+        self.assertFalse(wcsAlmostEqualOverBBox(self.wcs, self.exposure.getWcs(), self.bbox))
+
+        desiredWcs = self.makeDesiredDistortedWcs()
+        self.assertWcsAlmostEqualOverBBox(desiredWcs, self.exposure.getWcs(), self.bbox)
+
+    def makeMinimalIsrConfig(self):
+        """Return an IsrConfig with all boolean flags disabled"""
+        isrConfig = IsrTask.ConfigClass()
+        for name in isrConfig:
+            if name.startswith("do"):
+                setattr(isrConfig, name, False)
```

Comment:
Is it guaranteed that all boolean flags will always start with "do"?

Reply:
I think it is guaranteed that all flags that control whether an operation is to be performed will start with `do`, and that is what matters here. The idea is to minimally process the data.

Reply:
I don't think it matters in this case. If the test passes, great. If not, the test an easily be updated to add any new flags that control which stages get performed. In any case, the rule need only apply to IsrTask for this test, and self-consistency within one task is more likely that from task to task. It certainly doesn't rise to the level of an RFC.